ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js

```diff
@@ -71,10 +71,10 @@ function generateAnyGeoFilter(property, model) {
             };
         case 'POLYGON':
             return {
-                type: 'INTERSECTS',
+                type: model.polygonBufferWidth > 0 ? 'DWITHIN' : 'INTERSECTS',
                 property: property,
-                value: 'POLYGON' +
-                sanitizeForCql(JSON.stringify(polygonToCQLPolygon(model.polygon)))
+                value: `POLYGON${sanitizeForCql(JSON.stringify(polygonToCQLPolygon(model.polygon)))}`,
+                ...(model.polygonBufferWidth && {distance: DistanceUtils.getDistanceInMeters(model.polygonBufferWidth, model.polygonBufferUnits)})
```

Comment:
❓ Do we need to do the check here?

Reply:
Are you asking whether we need this guard at all or whether it should go up a couple of lines into it's own block?